tests/unit/Core/Batch/BatchJobTest.php

```diff
@@ -49,25 +49,25 @@ public function testCustom()
                 'batchSize' => 1000,
                 'callPeriod' => 1.0,
                 'bootstrapFile' => __FILE__,
-                'workerNum' => 10
+                'numWorkers' => 10
             )
         );
         $this->assertEquals(1000, $job->getBatchSize());
         $this->assertEquals(1.0, $job->getCallPeriod());
-        $this->assertEquals(10, $job->getWorkerNum());
-        $this->assertEquals(__FILE__, $job->getBootstrapFile());
-        $this->assertEquals(1, $job->getIdNum());
-        $this->assertEquals('testing', $job->getIdentifier());
+        $this->assertEquals(10, $job->numWorkers());
+        $this->assertEquals(__FILE__, $job->bootstrapFile());
+        $this->assertEquals(1, $job->id());
+        $this->assertEquals('testing', $job->identifier());
     }
 
-    public function testRun()
-    {
-        $job = new BatchJob('testing', array($this, 'runJob'), 1);
-        $items = array('apple', 'orange', 'banana');
-        $expected = array('APPLE', 'ORANGE', 'BANANA');
-        $job->run($items);
-        $this->assertEquals($expected, $this->items);
-    }
+    // public function testRun()
```

Comment:
Why is this commented out?

Reply:
Ah, this test should have been removed. The semantics of the `JobInterface#run()` is now for it to be the loop that runs forever. `BatchJob#run()` now contains the looping logic (reading from the sysv message queue) that was previously in the `BatchDaemon`.

Reply:
It's possible that we can abstract out the message queue interface and mock it here to allow testing that the `BatchJob` delegates items from the message queue into configured `callable` item handler.

Reply:
Ok, then let's just remove?